compiler/src/dotty/tools/dotc/typer/Inliner.scala

```diff
@@ -337,6 +337,22 @@ object Inliner {
     def codeOf(arg: Tree, pos: SrcPos)(using Context): Tree =
       Literal(Constant(arg.show)).withSpan(pos.span)
   }
+
+  extension (tp: Type) {
+    /** same as widenTermRefExpr, but preserves modules and singleton enum values */
+    private final def widenInlineScrutinee(using Context): Type = tp.stripTypeVar match {
+      case tp as ModuleOrEnumValueRef()    => tp
+      case tp: TermRef if !tp.isOverloaded => tp.underlying.widenExpr.widenInlineScrutinee
```

Comment:
The pattern makes it unclear that we are matching twice on a `TermRef`. Maybe refactor it to

```scala
private final def widenInlineScrutinee(using Context): Type = tp.stripTypeVar match {
  case tp: TermRef  => 
    val sym = tp.termSymbol
    if sym.isAllOf(EnumCase, butNot=JavaDefined) || sym.is(Module) then tp
    else if !tp.isOverloaded then tp.underlying.widenExpr.widenInlineScrutinee
    else tp
  case _ => tp
}   
```

Reply:
thank you for the suggestion, I have implemented it